### DISCUSSION TITLE
How can I validate if a `MimeMessage` is valid and can be delivered?

### DISCUSSION THREAD

**Discussion by ramonsmits** [jstedfast/MailKit#1762] [Q&A]
How can I validate if a `MimeMessage` is valid and has a minimum set of required properties set for it to be delivered?

**Comment by jstedfast** [comment]
RFCs like [rfc5322](https://tools.ietf.org/html/rfc5322) suggest that the only required headers are `From`, `Date` and `To`, `Cc`, and/or `Bcc` (in other words, you need the sender, date, and 1 or more recipients to be defined). Older versions of the specification seemed to indicate that `Message-Id` *should* be present as well.

MimeKit automatically sets a `Date` and `Message-Id` header when you create a `new MimeMessage`.
